flac.go

```diff
@@ -253,7 +302,113 @@ func (stream *Stream) ParseNext() (f *frame.Frame, err error) {
 	return frame.Parse(stream.r)
 }
 
-// Seek is not implement yet.
-func (stream *Stream) Seek(offset int64, whence int) (read int64, err error) {
-	return 0, nil
+// Seek to offset where offset represents the sample number in the flac file.
+//
+// TODO: return an error if sample is invalid?  Do nothing?
+//
+// sample could be invalid if:
+// whence == io.SeekEnd && sample is not negative
+// whence == io.SeekStart && sample is negative
+// whence == io.SeekCurrent and sample + current sample < 0 or > stream.Info.NSamples
+func (stream *Stream) Seek(sample int64, whence int) (read int64, err error) {
+	if stream.seekTable == nil {
+		if err := stream.makeSeekTable(); err != nil {
+			return 0, err
+		}
+	}
+
+	rs := stream.r.(io.ReadSeeker)
+
+	var point meta.SeekPoint
+	switch whence {
+	case io.SeekStart:
+		point = stream.searchFromStart(sample)
+	case io.SeekCurrent:
+		o, err := rs.Seek(0, io.SeekCurrent)
+		if err != nil {
+			return 0, err
+		}
+		point = stream.searchFromCurrent(sample, o)
+	case io.SeekEnd:
+		point = stream.searchFromEnd(sample)
+	default:
+		return 0, ErrInvalidSeek
+	}
+
+	_, err = rs.Seek(stream.dataStart+int64(point.Offset), io.SeekStart)
+	return int64(point.SampleNum), err
+}
+
+func (stream *Stream) searchFromCurrent(sample, offset int64) (p meta.SeekPoint) {
+	for _, p = range stream.seekTable.Points {
+		if int64(p.Offset) >= offset {
+			return stream.searchFromStart(int64(p.SampleNum) + sample)
+		}
+	}
+	return p
+}
+
+func (stream *Stream) searchFromEnd(sample int64) (p meta.SeekPoint) {
+	return stream.searchFromStart(int64(stream.Info.NSamples) + sample)
+}
+
+func (stream *Stream) searchFromStart(sample int64) (p meta.SeekPoint) {
+	for _, p = range stream.seekTable.Points {
+		if p.SampleNum >= uint64(sample) {
+			return p
+		}
+	}
+	return p
```

Comment:
It wasn't immediately clear to me that this line returned the zero value. For clarity, consider returning `meta.SeekPoint{}` which stands out as a zero value, and thus not the "intended" code path.

Then, the function would also be updated to not used named returned values. While this function is not exported it doesn't matter too much. For exported API however, [one suggestion](https://github.com/golang/go/wiki/CodeReviewComments#named-result-parameters) is to only used named returns when they help support documentation, e.g. when a function returns two integers `(x, y int)`.

Reply:
This would return the last SeekPoint.  I've been playing with ReadSeeker a bit and returning the last point actually seems to mimic the behavior seen here:  

https://play.golang.org/p/WStOyCU2wU9

In other words, seeking before the start isn't an error seeking past the end isn't either.